### PR TITLE
update sqlalchemy

### LINKED ISSUE #79: Upgrade to SQLalchemy >2.0
## Detailed Description
This can be done using https://github.com/openclimatefix/pv-site-datamodel/issues/88
and pv-site-datamodel==1.0.17

## Context
- sqlalmacy upgrades should make things faster

## Possible Implementation
upgrade requirements and run tests

### PR DESCRIPTION
**Pull request by rahul-maurya11b**
# Pull Request

Fixes #79 

## How Has This Been Tested?

ran make lint and fixed errors to pass the tests

- [ ] Yes

_If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_

- [ ] Yes

## Checklist:

- [ ] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by rahul-maurya11b** [comment]
@peterdudfield sir this fails some tests , I ran make lint in forecast_inference and database_cleanup and fixed errors based on error result but for that I have to set some values to None not sure of that and also it passed locally but fails after pushing gives error : `The currently activated Python version 3.10.12 is not supported by the project (^3.11).` to solve this I changed  some versions to 3.11 still issue persist 
If  I am running pytest using poetry run pytest tests it shows timeout no test ran .. please check what changes I should make?

**Comment by peterdudfield** [comment]
Could you update to the most recent branch, and then the CI will run. I can then have a look at the failing tests

File: forecast-inference/forecast_inference/app.py [hunk 1]
```diff
@@ -50,6 +50,8 @@ def _run_model_and_save_for_one_pv(
             return False
 
     site_uuid = UUID(pv_id)
+    print('written just to remove lint error testing purpose')
```
Comment (peterdudfield) [inline forecast-inference/forecast_inference/app.py:53]: can we rmoeve this print statment?

File: forecast-inference/forecast_inference/app.py [hunk 2]
```diff
@@ -66,7 +68,7 @@ def _run_model_and_save_for_one_pv(
         with profile(f'Writing {len(rows)} forecast values to db for pv "{pv_id}"'):
             with database_connection.get_session() as session:
                 forecast = ForecastSQL(
-                    site_uuid=site_uuid, forecast_version="0.0.0", timestamp_utc=timestamp
+                    site_uuid=None, forecast_version="0.0.0", timestamp_utc=timestamp
```
Comment (peterdudfield) [inline forecast-inference/forecast_inference/app.py:71]: is this right?

**Comment by peterdudfield** [comment]
Thank you, ive included these changes in #84. Ill close this for now

**Pull request closed without merging**